### PR TITLE
feat(commands): implement SIP-1.5 Tauri register_account command

### PR DESCRIPTION
**Pull request by pcamarajr**
## Summary

Implements SIP-1.5: Tauri `register_account` command with comprehensive input validation, error handling, and integration tests. This PR adds three new Tauri commands for SIP account registration management and establishes the application state management pattern for future commands.

## Changes

- **New AppState structure** (`src-tauri/src/state.rs`): Thread-safe wrapper for `AuthService` using `Arc<Mutex<>>` pattern
- **Auth commands module** (`src-tauri/src/commands/auth.rs`): Three new Tauri commands:
  - `register_account`: Registers SIP account with full validation
  - `get_registration_status`: Returns current registration state
  - `unregister_account`: Unregisters SIP account
- **Extended validation functions** (`src-tauri/src/commands/validation.rs`):
  - `validate_port_range()`: Validates port numbers (1-65535)
  - `validate_protocol()`: Validates transport protocol (udp/tcp/tls)
  - `validate_contact_uri()`: Basic SIP URI format validation
- **Error conversion** (`src-tauri/src/domain/errors.rs`): Implemented `From<SipError> for CommandError` for seamless error handling
- **Command registration**: Updated `lib.rs` to initialize AppState and register all auth commands
- **Integration tests**: Comprehensive test suite covering all validation scenarios
- **Documentation updates**: Marked SIP-1.5 as completed in roadmap and README

### How to Test

1. **Setup**:
   ```bash
   git checkout feat/sip-1-5-tauri-command
   npm install
   cd src-tauri && cargo build
   ```

2. **Test Steps**:

   - **Validation Tests**: Run `cargo test --test register_account_integration_test` to verify all validation functions
   - **Unit Tests**: Run `cargo test --lib` to verify all 165 unit tests pass
   - **Manual Testing** (requires frontend integration):
     - Call `register_account` with valid credentials
     - Call `get_registration_status` to check state
     - Call `unregister_account` to unregister
     - Test validation errors with invalid inputs (empty fields, invalid port, invalid protocol)

3. **Expected Behavior**:
   - All validation functions reject invalid inputs with appropriate error messages
   - Commands return proper `CommandError` types for frontend handling
   - Registration state transitions work correctly through AuthService
   - All 165 existing tests continue to pass

### Test Coverage

- [x] Unit tests added/updated (validation functions have comprehensive tests)
- [x] Integration tests added/updated (`register_account_integration_test.rs`)
- [ ] Manual testing completed (requires frontend integration - SIP-1.6)
- [x] Edge cases tested (empty strings, invalid formats, boundary values)

### Code Quality

- [x] Code follows project conventions
- [x] Functions are focused and single-purpose
- [x] Variable names are descriptive
- [x] No code duplication
- [x] Error handling is appropriate

### Framework Compliance

- [x] Svelte 5 runes mode (no legacy syntax) - N/A (backend only)
- [x] Rust code follows idiomatic patterns
- [x] Tauri IPC follows best practices
- [x] Design system components used correctly - N/A (backend only)

### Security

- [x] No hardcoded secrets
- [x] Input validation present (comprehensive validation for all inputs)
- [x] Sensitive data handled securely (passwords passed through, not logged)
- [x] Tauri capabilities properly configured

### Documentation

- [x] README.md updated (marked SIP-1.5 as completed)
- [x] Code comments explain complex logic
- [x] Architecture docs updated (roadmap updated)

## Breaking Changes

None - This is a new feature addition. Existing commands (`greet`) remain unchanged.

## Related Issues

Implements **SIP-1.5** from the implementation roadmap (`docs/architecture/05-implementation-roadmap.md`).

## Additional Notes

- The `register_account` command currently generates a default contact URI if not provided. Future enhancements could use the actual local IP address from the SIP client.
- Full end-to-end registration testing requires a test SIP server (Asterisk, FreeSWITCH, or mock server) - this is documented in the integration test file.
- The AppState pattern established here will be used for future commands that need access to services.
- All validation functions follow the existing pattern from `validation.rs` and return `Result<(), CommandError>` for use with the `?` operator.

### CODE REVIEW

File: src-tauri/src/commands/auth.rs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+};
+use crate::domain::entities::credentials::{Credentials, TransportProtocol};
+use crate::domain::entities::registration::RegistrationState;
+use crate::domain::errors::CommandError;
+use crate::state::AppState;
+use std::net::ToSocketAddrs;
+use tauri::State;
+
+/// Convert protocol string to TransportProtocol enum
+fn parse_protocol(protocol: &str) -> Result<TransportProtocol, CommandError> {
```
Comment (pcamarajr) [inline src-tauri/src/commands/auth.rs]: **⚠️ Warning:** Code duplication - `parse_protocol` duplicates logic from `validate_protocol`

This function performs the same validation as `validate_protocol` in `validation.rs`. Since `validate_protocol` is already called on line 74, the validation in `parse_protocol` is redundant. Consider refactoring to reuse `validate_protocol` or extract the protocol-to-enum conversion logic into a shared function.

**Suggested fix:**
```rust
// After validate_protocol(&protocol)?; on line 74
let transport_protocol = match protocol.to_lowercase().as_str() {
    "udp" => TransportProtocol::Udp,
    "tcp" => TransportProtocol::Tcp,
    "tls" => TransportProtocol::Tls,
    _ => unreachable!(), // Already validated above
};
```

This eliminates the duplicate `parse_protocol` function entirely.

File: src-tauri/src/commands/auth.rs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+    // Build credentials
+    let credentials =
+        Credentials::new(server.clone(), port, transport_protocol, username, password);
+
+    // Resolve server address
+    let server_addr = resolve_server_address(&server, port)?;
+
+    // Generate contact URI if not provided
+    let contact = contact_uri.unwrap_or_else(|| {
```
Comment (pcamarajr) [inline src-tauri/src/commands/auth.rs]: **⚠️ Warning:** Contact URI generation may be incorrect

The default contact URI uses `server` as the host, but SIP contact URIs should typically use the local IP address of the client, not the server address. This could cause registration issues with some SIP servers.

**Current behavior:** `sip:username@server:port`
**Expected behavior:** `sip:username@local_ip:local_port`

**Reference:** The comment on line 92 mentions "For now, use a simple format - in production, get actual local IP", which suggests this is a known limitation. Consider documenting this as a TODO or adding a note that this will be addressed in a future PR.

File: src-tauri/src/lib.rs [hunk 3]
```diff
@@ -12,13 +12,35 @@ pub mod services;
...
+use tauri::Manager;
 
 #[cfg_attr(mobile, tauri::mobile_entry_point)]
 pub fn run() {
     tauri::Builder::default()
         .plugin(tauri_plugin_opener::init())
-        .invoke_handler(tauri::generate_handler![greet])
+        .setup(|app| {
+            // Initialize SIP client and AppState
+            let rt = tokio::runtime::Runtime::new().expect("Failed to create tokio runtime");
```
Comment (pcamarajr) [inline src-tauri/src/lib.rs]: **🚫 Blocker:** Panic on initialization failure

Using `expect()` will cause the application to panic if the Tokio runtime creation fails. This is a critical failure point that should be handled gracefully.

**Suggested fix:**
```rust
let rt = tokio::runtime::Runtime::new()
    .map_err(|e| format!("Failed to create tokio runtime: {}", e))?;
```

However, since `setup()` returns `Result<(), Box<dyn std::error::Error>>`, you can use `?` operator instead of `expect()`.

**Review by pcamarajr — COMMENTED** [review]
## Code Review Summary

✅ **Overall Assessment:** The PR implements SIP-1.5 with comprehensive validation, error handling, and integration tests. The code structure follows good practices, but there are critical issues with error handling in the initialization code that should be addressed.

**Issues Found:**
- 🚫 Blockers: 2
- ⚠️ Warnings: 2
- 💡 Suggestions: 0

**Key Findings:**
- **Critical:** `src-tauri/src/lib.rs:29` and `src-tauri/src/lib.rs:31` use `expect()` which will cause application panics on initialization failure. These should be replaced with proper error handling.
- **Code Quality:** `src-tauri/src/commands/auth.rs:15` has code duplication between `parse_protocol` and `validate_protocol`. This violates DRY principle and should be refactored.
- **Functionality:** `src-tauri/src/commands/auth.rs:90` - Contact URI generation uses server address instead of local IP, which may cause registration issues. This is documented as a known limitation but should be noted.

**Review Details:**
- Inline comments posted for line-specific issues
- Category review comments posted above for:
  - Code Quality & Maintainability (1 warning)
  - Rust & Tauri Compliance (2 blockers)
- Categories reviewed: Functionality & Correctness ✅, Code Quality & Maintainability ⚠️, Security ✅, Architecture Compliance ✅, Svelte 5 Framework Compliance ✅ (N/A - backend only), Rust & Tauri Compliance 🚫, Design System & UI Compliance ✅ (N/A - backend only), Testing & Documentation ✅, Performance & Optimization ✅, README & Documentation Updates ✅

**Positive Aspects:**
- Comprehensive input validation for all command parameters
- Well-structured error handling with proper error type conversions
- Good test coverage with integration tests
- README and roadmap documentation updated correctly
- Follows established patterns from existing codebase
- Proper use of async/await and Tokio runtime

**Recommended Actions:**
1. Replace `expect()` calls in `lib.rs` with proper error handling using `?` operator
2. Refactor `parse_protocol` to eliminate code duplication with `validate_protocol`

**Optional Improvements:**
- Consider documenting the contact URI limitation more explicitly (e.g., with a TODO comment or issue reference)
- Consider adding more specific error messages for address resolution failures

File: src-tauri/src/lib.rs [hunk 4]
```diff
@@ -12,13 +12,35 @@ pub mod services;
...
+            let client = rt
+                .block_on(SipClient::new_udp_any())
```
[the remark this answers is not in the document]
Comment (pcamarajr) [reply src-tauri/src/lib.rs]: ✅ Resolved in commit dcff5d9d14755f769f9d08eaddf87fa432e641f1
